Updating logic to check if a list of observables is qwc (#2798)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

  [(#2721)](https://github.com/PennyLaneAI/pennylane/pull/2721)

* Added operation `qml.QutritUnitary` for applying user-specified unitary operations on qutrit devices.
  [(#2699)](https://github.com/PennyLaneAI/pennylane/pull/2699)  

> **Operator Arithmetic:**

Author: There were two **Operator Arithmetic** headers, I just removed one

  labels.
  [(#2779)](https://github.com/PennyLaneAI/pennylane/pull/2779)

* Adds a new function to compare operators. `qml.equal` can be used to compare equality of parametric operators taking 

>   into account their interfaces and trainability.

Author: This was spilling over multiple lines, so cleaned it up

  DeviceArray([-0.78849435, -0.8287073 , -0.85608006], dtype=float32)
  ```

> * Adds an `are_qwc_pauli_words(lst_pauli_words)` function which checks if a list of 
>   pauli words is qubit-wise commuting. This improves performance when measuring hamiltonians 

Reviewer: ```suggestion
* Added an `are_pauli_words_qwc` function which checks if certain 
  Pauli words are pairwise qubit-wise commuting. This new function improves performance when measuring hamiltonians 
```
